apps/super_simplify/reduction_order.cpp

```diff
@@ -65,18 +65,153 @@ std::set<std::string> find_divisors(const Expr &e) {
     return d.divisors;
 }
 
-bool check_divisors(Expr &LHS, Expr &RHS) {
-  // check that all divisors on RHS appear as divisors on LHS
-  std::set<std::string> lhs_divisors = find_divisors(LHS);
-  std::set<std::string> rhs_divisors = find_divisors(RHS);
-  for (auto const& rhs_term : rhs_divisors) {
-    if (lhs_divisors.count(rhs_term) == 0) {
-      return false;
+class VectorOpCount : public IRVisitor {
+    void visit(const Ramp *op) override {
+        counter += 1;
+    }
+    void visit(const Broadcast *op) override {
+        counter += 1;
+    }
+public:
+    int counter = 0;
+};
+
+int get_vector_count(const Expr &e) {
+    VectorOpCount rcounter;
+    e.accept(&rcounter);
+    return rcounter.counter;
+}
+
+bool check_divisors(const Expr &LHS, const Expr &RHS) {
+    // check that all divisors on RHS appear as divisors on LHS
+    std::set<std::string> lhs_divisors = find_divisors(LHS);
+    std::set<std::string> rhs_divisors = find_divisors(RHS);
+    for (auto const& rhs_term : rhs_divisors) {
+        if (lhs_divisors.count(rhs_term) == 0) {
+            return false;
+        }
     }
-  }
   return true;
 }
 
+class NonlinearOpsCount : public IRVisitor {
+    void visit(const Div *op) override {
+        counter += 1;
+        op->a.accept(this);
+        op->b.accept(this);
+    }
+    void visit(const Mod *op) override {
+        counter += 1;
+        op->a.accept(this);
+        op->b.accept(this);
+    }
+    void visit(const Mul *op) override {
+        counter += 1;
+        op->a.accept(this);
+        op->b.accept(this);
+    }
+public:
+    int counter = 0;
+};
+
+int get_nonlinear_op_count(const Expr &e) {
+    NonlinearOpsCount nl;
+    e.accept(&nl);
+    return nl.counter;
+}
+
+class IsConstant : public IRVisitor {
+    void visit(const Variable *op) override {
+        if (op->name[0] == 'c') {
```

Comment:
Is this some convention specific to this app?

Reply:
Yeah, it's a bit of a hack, but note that this pr is not into master